### PR TITLE
Play Lab now uses JSInterpreter instead of Interpreter directly

### PR DESCRIPTION
**Pull request by cpirich**
- moved `selectEditorRowCol` to codegen.js so it can be shared by all apps
- created studio/apiJavascript.js to expose studio APIs in the form we want to be used for droplet/ace. This is a subset of the full studio API set (no long running commands). The API signatures currently differ only in that they don't take the `id` parameter to record the blockly blockId first.
- updated studio/dropletConfig.js to use the new `apiJavascript` as the parent for each API function we expose in the interpreter sandbox.
- moved `SpriteSpeed` and `SpriteSize` constants from studio/api.js to studio/constants.js. Expose them again through studio/api.js so generated blockly code can reference those constants.
- added temporary `handleExecutionError` and `outputError` functions to studio.js. We need to sort through how this app wants to handle parsing and runtime errors.
- changed studio (playlab) to use the new JSInterpreter object. Removed code from studio that is no longer needed with this change.

### CODE REVIEW

File: apps/src/codegen.js [hunk 1]
```diff
@@ -451,6 +451,32 @@ exports.isAceBreakpointRow = function (session, userCodeRow) {
...
 
+/**
+ * Selects code in droplet/ace editor.
+ *
+ * This function simply highlights one spot, not a range. It is typically used
+ * to highlight where an error has occurred.
+ */
+exports.selectEditorRowCol = function (editor, row, col) {
+  if (editor.currentlyUsingBlocks) {
+    var style = {color: '#FFFF22'};
```
Comment (breville) [inline apps/src/codegen.js]: worth moving this into CSS styling at some point?

File: apps/src/studio/apiJavascript.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+exports.setSpriteSpeed = function (spriteIndex, value) {
+  Studio.queueCmd(null, 'setSpriteSpeed', {
+    'spriteIndex': spriteIndex,
+    'value': value
+  });
+};
+
+/*
+exports.setSpriteSize = function (spriteIndex, value) {
```
Comment (breville) [inline apps/src/studio/apiJavascript.js]: why is this commented out?

File: apps/src/studio/apiJavascript.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+exports.setSpritePosition = function (spriteIndex, value) {
+  Studio.queueCmd(null, 'setSpritePosition', {
+    'spriteIndex': spriteIndex,
+    'value': value
+  });
+};
+
+/*
+exports.setSpriteXY = function (spriteIndex, xpos, ypos) {
```
Comment (breville) [inline apps/src/studio/apiJavascript.js]: ditto for the others commented out..

File: apps/src/studio/dropletConfig.js [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
-  {'func': 'move', 'category': 'Play Lab', 'params': ["0", "1"] },
-  {'func': 'playSound', 'category': 'Play Lab', 'params': ["'slap'"] },
-  {'func': 'changeScore', 'category': 'Play Lab', 'params': ["1"] },
-  {'func': 'setSpritePosition', 'category': 'Play Lab', 'params': ["0", "7"] },
-  {'func': 'setSpriteSpeed', 'category': 'Play Lab', 'params': ["0", "8"] },
-  {'func': 'setSpriteEmotion', 'category': 'Play Lab', 'params': ["0", "1"] },
-  {'func': 'throwProjectile', 'category': 'Play Lab', 'params': ["0", "1", "'blue_fireball'"] },
-  {'func': 'vanish', 'category': 'Play Lab', 'params': ["0"] },
-  {'func': 'onEvent', 'category': 'Play Lab', 'params': ["'when-left'", "function() {\n  \n}"] },
+  {'func': 'setSprite', 'parent': api, 'category': 'Play Lab', 'params': ["0", "'cat'"] },
```
Comment (breville) [inline apps/src/studio/dropletConfig.js]: are these skinnable for skins where it's not "cat" but something else?

**Comment by breville** [comment]
new code for me, but looks good.  just a few questions in comments above.

File: apps/src/studio/dropletConfig.js (hunk 4 above)
Comment (cpirich) [inline apps/src/studio/dropletConfig.js]: yes, in the sense that level builders can override the parameters for these blocks on a per-level basis.

File: apps/src/studio/apiJavascript.js (hunk 3 above)
Comment (cpirich) [inline apps/src/studio/apiJavascript.js]: Both are APIs that we could choose to enable (not long running blocks), but are not currently in the toolbox for the droplet level. I should make that more clear in my next change.

File: apps/src/codegen.js (hunk 1 above)
Comment (cpirich) [inline apps/src/codegen.js]: Unfortunately, droplet's API takes a parameter called `style` which is a object that contains a property called color. But not a real CSS style or class. We could move to at least create a constant in our code for this color, though.

**Pull request merged**